scss/themes/platform/_common/inc/_mixins.scss

```diff
@@ -144,33 +149,33 @@
 /// everything required for mobile navi
 /// @param {string} $menuActiveBg
 /// @param {string} $menuActiveText
-@mixin theme-mobile-navi ($menuActiveBg, $menuActiveText) {
-    @if($menuActiveBg != '' and $menuActiveText != '') {
+@mixin theme-mobile-navi ($menuActiveBg, $menuActiveText, $headerBg) {
+    @if($headerBg != '' and $menuActiveText != '') {
         .dark-bar {
             nav {
                 .settings-menu{
                     & > ul {
-                        @include submenu($menuActiveBg);
+                        @include submenu ($menuActiveBg, $menuActiveText, $headerBg);
                     }
                     &:hover {
                         .reduced-menu-trigger {
-                            background: colorVariation($menuActiveBg, 'menu', 'hover-bg');
+                            background: colorVariation($headerBg, 'menu', 'hover-bg');
                         }
                     }
                     .reduced-menu-trigger {
-                        text-shadow: 1px 1px 1px autoTextShadowColor($menuActiveBg, $menuActiveText);
+                        text-shadow: 1px 1px 1px autoTextShadowColor($headerBg, $menuActiveText);
                         padding: 13px 10px 11px;
                         color: colorVariation($menuActiveText, 'menu', 'text');
                         &:hover {
-                            background: colorVariation($menuActiveBg, 'menu', 'bg');
-                            border-color: colorVariation($menuActiveBg, 'menu', 'bg');
+                            background: rgba(255,255,255,0.55) !important;
```

Comment:
I have some doubts here: ` rgba(255,255,255,0.55)` is a good value for lighter brandings, but will probably fail on darker ones.
I suggest you do the following:
- In your real project set the color for `mainBarBg` testwise to a very dark color and compile
- observe the submenu on the right hand side as well the one for small screens
- check the hover effect, it might be in an undesired color
As a proper fix you could either work on the `variations` in the color directory or write a function that calculates the value based on the fact whether the background is light or dark.

Reply:
changed to 
background: colorVariation($menuActiveBg, 'menu', 'hover-bg') !important;